File: Java_Programming/SingleLevelX.java
class Base 
{
    public int i;
    public int j;

    public void Base()
    {
        System.out.println("Inside Base Constructor");
    }
    public void fun ()
    {
        System.out.println("Inside Base Fun");
    }

    public void gun ()
    {
        System.out.println("Inside Base Gun");
    }
}

class Derived extends Base
{
    public int x, y;

    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }

    public void sun ()
    {
        System.out.println("Inside Derived Sun");
    }
}


class SingleLevelX 
{
    public static void main(String A[]) 
    {
        Derived Dobj = new Derived();

        Dobj.fun();
        Dobj.gun();
        Dobj.sun();   
    }
}

File: C++_Programming/SingleLevelX.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int i, j;

        Base()
        {
            cout<<"Inside Base Constructor\n";
        }

        ~Base()
        {
            cout<<"Inside Base Destructor\n";
        }

        void fun()
        {
            cout<<"Inside Base Fun\n";
        }

        void gun()
        {
            cout<<"Inside Base Gun\n";
        }

};

class Derived : public Base
{
    public:
        int x, y;

        Derived()
        {
            cout<<"Inside Derived Constructor\n";
        }

        ~Derived()
        {
            cout<<"Inside Derived Destructor\n";
        }

        void sun ()
        {
            cout<<"Inside Derived Sun\n";
        }
};

int main ()
{
    Derived Dobj;

    
    Dobj.fun();
    Dobj.gun();
    Dobj.sun();

    return 0;
}